neuron/src/lib/Neuron/Zettelkasten/Graph.hs

```diff
@@ -48,9 +49,12 @@ backlinkForest conn z =
     . G.bfsForestBackwards z
     . G.induceOnEdge (== Just conn)
 
-backlinks :: Connection -> Zettel -> ZettelGraph -> [Zettel]
+backlinks :: Maybe Connection -> Zettel -> ZettelGraph -> [(Connection, Zettel)]
 backlinks conn z g =
-  G.preSetWithEdgeLabel (Just conn) z g
+  let includeConn = maybe isJust (const (== conn)) conn
+      g' = LAM.transpose $ G.getGraph $ G.induceOnEdge includeConn g
+      ns = Map.toList $ Map.findWithDefault mempty (G.vertexID z) $ LAM.adjacencyMap g'
+   in mapMaybe (\(v, e) -> (,) <$> e <*> G.findVertex v g) ns
```

Comment:
This implementation should be moved to `Data.Graph.Labelled.Algorithm`. You can just put them in the existing `preSetWithEdgeLabel` function. 

Also, you should use `getVertex` here instead of `findVertex`. The former works with the guarantee that the vertex exists -- which is the case here -- and doesn't have to deal with Maybe. This way you also don't have to ignore the error via the Applicative lift.

Aside: Maybe we should just pass the filter function directly -- i.e., `(Maybe Connection -> Bool)` instead `Maybe Connection` -- but it is okay if that's not done in this PR>

Reply:
> maybe isJust (const (== conn)) conn

Isn't this the same as `maybe isJust (==) conn`?

Reply:
I can move about half of the implementation into `Data.Graph.Labelled.Algorithm` due to `preSetWithEdgeLabel` being polymorphic in the edge type.

In `backlinks` the edge type is specialized to `Maybe Connection` which is why I post process the `postSet` by removing `Maybe` from the edges.  The other issue with moving more of the code into `preSetWithEdgeLabel` is that `LAM.traspose` imposes more constraints on the edge type than currently exist.  For example, `e` would need to be an instance of `Monoid` and `Ord`.

How does that sound to you?

Reply:
`preSetWithEdgeLabel` can just take those extra constraints on `e`. If you pass `(e -> Bool)` instead of `Maybe Connection` (as the filter) to it, the function can remain polymorphic - and you can then move all of the implementation.

Reply:
Most of the logic has been moved into `preSetWithEdgeLabel`.  Please let me know what you think.

Reply:
@pjones Almost there. The `backlinks` function should take `(Maybe Connection -> Bool)`  instead of `Maybe Connection` - just to avoid potential confusion.

(`Maybe Connection` is used as the edge type only because the alga type requires it to be a monoid - where the empty value signifies "no connection"; using the same type for the filter can be confusing).

Reply:
Good idea.  The latest commit has this change.

Reply:
Looks good to me. Once you mark it non-WIP, I'll merge it.

Reply:
Done.